moxy-compiler/src/main/java/moxy/compiler/viewstate/ViewInterfaceProcessor.kt

```diff
@@ -236,13 +237,42 @@ class ViewInterfaceProcessor(
         return modifiers.contains(Modifier.STATIC)
     }
 
-    private inline fun <T, R> List<T>.flatMapToSet(transform: (T) -> Iterable<R>): Set<R> {
-        return flatMapTo(LinkedHashSet(), transform)
-    }
-
     companion object {
         private const val OPTION_DEFAULT_STRATEGY = MvpCompiler.DEFAULT_MOXY_STRATEGY
         private val DEFAULT_STATE_STRATEGY: TypeElement =
             elementUtils.getTypeElement(AddToEndSingleStrategy::class.java.canonicalName)
     }
+
+    class StockViewMethod(
+            val targetInterfaceElement: DeclaredType,
+            val element: ExecutableElement,
+            val strategy: TypeElement?,
+            val tag: String
+    )
+
+    private fun StockViewMethod.toViewMethod(): ViewMethod {
+        if(strategy == null) throw IllegalStateException("Strategy can't be null")
+        return ViewMethod(targetInterfaceElement, element, strategy, tag)
+    }
+
+    private fun TypeElement.isNotMvpViewExtend(): Boolean {
+        return this.interfaces.find { parent ->
+            if ((parent as DeclaredType).toString() == MvpView::class.java.canonicalName) true
+            else !parent.asTypeElement().isNotMvpViewExtend()
+        } == null
+    }
+
+    private fun List<StockViewMethod>.validateAndMap(viewInterface: TypeElement): Set<ViewMethod> {
+        val methods = this
+        return filter { method ->
+            if (method.strategy == null && methods.none { it.element.simpleName == method.element.simpleName && it.strategy != null }) {
```

Comment:
Elements are compared by name. So, if the interface would have a method overload with the same name, compilation would fail. `ViewMethod` has an `equals` method to work around this problem
Also, there should be a test covering that case